### PR TITLE
Mgdapi 3185

### PR DESCRIPTION
**Pull request by den-rgb**
# Issue link
https://issues.redhat.com/browse/MGDAPI-3185 

# What
Added three new 3scale tests to verify the creation of Tenant, Backend and Product via CR.

# Verification steps
- Provision cluster
- Install RHOAM from https://github.com/integr8ly/integreatly-operator/pull/2409
- Run: 
``` 
INSTALLATION_TYPE=managed-api TEST=H08 make test/e2e/single BYPASS_STORAGE_TYPE_CHECK=true
```
```
INSTALLATION_TYPE=managed-api TEST=H09 make test/e2e/single BYPASS_STORAGE_TYPE_CHECK=true
```
```
INSTALLATION_TYPE=managed-api TEST=H10 make test/e2e/single BYPASS_STORAGE_TYPE_CHECK=true
```
- Will expect e2e on this pr to also pass.

### CODE REVIEW

**Comment by Patryk-Stefanski** [comment]
Can verify that H08-H10 tests pass locally and e2e test pass in prow, code looks good .
/lgtm

File: test/common/tests.go [hunk 1]
```diff
@@ -71,6 +71,9 @@ var (
 				/*FLAKY on RHMI*/ {"Verify Alerts are not firing during or after installation apart from DeadMansSwitch", TestIntegreatlyAlertsFiring},
 				{"Verify prometheus metrics scrapped", TestMetricsScrappedByPrometheus},
 				{"A27 + A28 - Verify pod priority class is created and set", TestPriorityClass},
+				{"H08 - Verify that backend can be created via backend CR", Test3scaleBackendViaCR},
+				{"H09 - Verify that product can be created via tenant CR", Test3scaleProductViaCR},
+				{"H10 - Verify that tenant can be created via tenant CR", Test3scaleTenantViaCr},
```
Comment (psturc) [inline test/common/tests.go:76]: We already [have test cases with these IDs](https://github.com/integr8ly/integreatly-operator/tree/master/test-cases/tests/products)

```suggestion
				{"H29 - Verify that backend can be created via backend CR", Test3scaleBackendViaCR},
				{"H30 - Verify that product can be created via tenant CR", Test3scaleProductViaCR},
				{"H31 - Verify that tenant can be created via tenant CR", Test3scaleTenantViaCr},
```

could we also add test cases for these automated tests [here](https://github.com/integr8ly/integreatly-operator/tree/master/test-cases/tests/products) and mark them as automated? Thanks!

(A guide for adding new test cases is [here](https://github.com/integr8ly/integreatly-operator/tree/master/test-cases#how-to-create-a-test-case))

**Comment by den-rgb** [comment]
@psturc

**Pull request merged**